extensions/html/server/src/modes/javascriptMode.ts

```diff
@@ -288,9 +288,13 @@ export function getJavascriptMode(documentRegions: LanguageModelCache<HTMLDocume
 	};
 };
 
-function convertRange(document: TextDocument, span: { start: number, length: number }): Range {
-	let startPosition = document.positionAt(span.start);
-	let endPosition = document.positionAt(span.start + span.length);
+function convertRange(document: TextDocument, span: { start: number | undefined, length: number | undefined }): Range {
```

Comment:
This is the main logic change. TS allows both the start and length to be undefined